### PR TITLE
 fix(proguard): Correctly merge symbolicated and unsymbolicated exceptions

### PR DESCRIPTION
**Pull request by loewenheim**
We don't send all exceptions to Symbolicator. This means that obviously we can't just zip the exceptions in the event and those we get back from Symbolicator because they don't correspond one-to-one.

I kinda hate how this is written right now and would be thankful for suggestions on how to simplify.

### CODE REVIEW

File: src/sentry/lang/java/processing.py [hunk 1]
```diff
@@ -253,26 +254,32 @@ def process_jvm_stacktraces(symbolicator: Symbolicator, data: Any) -> Any:
...
+    for e, p in zip(processable_exceptions, response["exceptions"]):
+        index = e.pop("index")
+        e["type"] = p["type"]
+        e["module"] = p["module"]
+        symbolicated_exceptions[index] = e
+
+    symbolicator_exceptions = []
+    for index, raw_exception in enumerate(all_exceptions):
+        if symbolicated := symbolicated_exceptions.get(index):
+            symbolicator_exceptions.append(symbolicated)
```
Comment (Swatinem) [inline src/sentry/lang/java/processing.py:270]: I understand the logic, but you might as well simplify it in this way:

- Return the "original" exception objects from `_get_exceptions_for_symbolication`, filtered for those that need symbolication
- Send only the relevant data of them to symbolicator using a `map`-like function (or list compresention in Python)
- Just iterate over the `zip` of the filtered list, plus the symbolicated result, and apply the modifications

As objects in python are refcounted, you can just modify the exception object you filtered directly, no need to construct yet another list here that just contains references to objects you never touched.

**Review by Swatinem — APPROVED** [review]

File: src/sentry/lang/java/processing.py [hunk 2]
```diff
@@ -81,14 +82,16 @@ def _handles_frame(frame: dict[str, Any], platform: str) -> bool:
     "Returns whether the frame should be symbolicated by JVM symbolication."
 
     return (
-        "function" in frame and "module" in frame and (frame.get("platform") or platform) == "java"
+        "function" in frame
+        and "module" in frame
+        and (frame.get("platform", None) or platform) == "java"
```
Comment (anonrig) [inline src/sentry/lang/java/processing.py:87]: This can be `frame.get("platform", platform)`

Comment (asottile-sentry) [reply src/sentry/lang/java/processing.py:87]: I suspect something sets `"platform": null`

File: pyproject.toml [hunk 3]
```diff
@@ -621,6 +621,7 @@ module = [
     "sentry.buffer.redis",
     "sentry.eventstore.reprocessing.redis",
     "sentry.issues.related.*",
+    "sentry.lang.java.processing",
```
Comment (asottile-sentry) [inline pyproject.toml:624]: nevermind!  read the wrong blocklist on an unrelated PR merge conflict

**Pull request merged**